drivers/can/can_stm32fd.c

```diff
@@ -74,25 +63,29 @@ static int can_stm32fd_clock_enable(const struct device *dev)
 	const struct can_stm32fd_config *stm32fd_cfg = mcan_cfg->custom;
 	const struct device *const clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
 
-	LL_RCC_SetFDCANClockSource(CAN_STM32FD_CLOCK_SOURCE);
-
-	/* LL_RCC API names do not align with PLL output name but are correct */
-#ifdef CONFIG_CAN_STM32FD_CLOCK_SOURCE_PLL1Q
-	LL_RCC_PLL1_EnableDomain_48M();
-#elif CONFIG_CAN_STM32FD_CLOCK_SOURCE_PLL2P
-	LL_RCC_PLL2_EnableDomain_SAI();
-#endif
+	if (IS_ENABLED(STM32_CANFD_DOMAIN_CLOCK_SUPPORT) && (stm32fd_cfg->pclk_len > 1)) {
+		ret = clock_control_configure(clk,
+				(clock_control_subsys_t)&stm32fd_cfg->pclken[1],
+				NULL);
+		if (ret < 0) {
+			LOG_ERR("Could not select can_stm32fd domain clock");
+			return ret;
+		}
+	}
 
 	if (!device_is_ready(clk)) {
 		return -ENODEV;
 	}
```

Comment:
This needs to be moved to before configuring the domain clock, right?

Reply:
Thanks!
Yes it should. While in the current clock control driver implementation it wouldn't really matter it definitely should be checked before being used.